teuthology/worker.py

```diff
@@ -133,7 +133,12 @@ def main(ctx):
         except SkipJob:
             continue
 
-        job.delete()
+        # This try/except block is to keep the worker from dying when
+        # beanstalkc throws a SocketError
+        try:
+            job.delete()
+        except Exception:
```

Comment:
I'm not sure which module this SocketError is coming from; just trying to get things working properly so I can stop babysitting